src/main/java/net/sf/jabref/gui/groups/TransferableEntrySelection.java

```diff
@@ -39,7 +39,8 @@
     public TransferableEntrySelection(List<BibEntry> list) {
         this.selectedEntries = list;
         selectedEntriesCiteKeys = String.join(",",
-                this.selectedEntries.stream().map(BibEntry::getCiteKey).collect(Collectors.toList()));
+                this.selectedEntries.stream().map(BibEntry::getCiteKeyOptional).filter(key -> key.isPresent())
```

Comment:
The filter and second map is not necessary, it should workt with   map(BibEntry::getCiteKeyOptional).orElse().collect...